### PR TITLE
Update debian files

### PR DESCRIPTION
**Pull request by sumukhatv**
Changes to files in debian directory to facilitate building sonic-buildimage along with restapi container.

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -6,7 +6,6 @@ Build-Depends: dh-exec (>=0.3), debhelper (>= 9)
 Standards-Version: 1.0.0
 
 Package: sonic-rest-api
-Architecture: any
+Architecture: amd64
```
Comment (lguohan) [inline debian/control]: is this amd64?

Comment (lguohan) [reply debian/control]: this package should be architecture indepenent

**Comment by lguohan** [comment]
can you provide brief description of the PR. what have you achieved in this pr.

**Comment by lguohan** [comment]
can you add a README.md file to describe how to build? check this example. https://github.com/Azure/sonic-telemetry/blob/master/README.md

File: debian/rules [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+	dh $@  --with autotools-dev
+
+# dh_make generated override targets
+# This is example for Cmake (See https://bugs.debian.org/641051 )
+#override_dh_auto_configure:
+#	dh_auto_configure -- \
+#	-DCMAKE_LIBRARY_PATH=$(DEB_HOST_MULTIARCH)
+
+override_dh_auto_configure:
+	dh_auto_configure --
```
Comment (lguohan) [inline debian/rules]: do we need to override the auto_configure?

File: debian/rules [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+
+override_dh_auto_install:
+	dh_auto_install --destdir=debian/sonic-rest-api
 
+#override_dh_strip:
+#	dh_strip --dbg-package=swss-dbg
```
Comment (lguohan) [inline debian/rules]: looks like this is not needed?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  1. `mkdir debs; cd debs`
+  2. `sudo wget https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-all/lastSuccessfulBuild/artifact/target/debs/*zip*/debs.zip`
+  3. `sudo unzip debs.zip`
+  
+### Build Rest-API
+  1. Modify `copy.sh` SDIR path to point to your downloaded debs from the previous step
+  2. Execute ./build.sh
+  3. The above should generate 2 Docker images, 1 which is used for local development on your VM and 1 which is used for deployment on a TOR
+  4. Run `docker images` to check if rest-api dockers were generated <br/>
+      		<pre>`REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE` <br/>
```
Comment (lguohan) [inline README.md]: use ``` ``` for multiline code block.

https://confluence.atlassian.com/bitbucketserver/markdown-syntax-guide-776639995.html

File: README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+## Getting Started
+### Fetch Dependencies
+  1. `mkdir debs; cd debs`
+  2. `sudo wget https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-all/lastSuccessfulBuild/artifact/target/debs/*zip*/debs.zip`
+  3. `sudo unzip debs.zip`
+  
+### Build Rest-API
+  1. Modify `copy.sh` SDIR path to point to your downloaded debs from the previous step
+  2. Execute ./build.sh
+  3. The above should generate 2 Docker images, 1 which is used for local development on your VM and 1 which is used for deployment on a TOR
```
Comment (anish-n) [inline README.md]: Rename TOR as router, here and below

File: debian/changelog [hunk 6]
```diff
@@ -1,6 +1,11 @@
+sonic-rest-api-server (1.0.1) stable; urgency=medium
```
Comment (anish-n) [inline debian/changelog]: This results in a package named sonic-rest-api_1.0.1_amd64.deb, we need to change the Dockerfiles to reflect the new package name for the build to succeed

File: README.md (hunk 5 above)
Comment (sumukhatv) [reply README.md]: you mean switch?

Comment (anish-n) [reply README.md]: Yes, either switch or router would work for it.

File: Makefile [hunk 7]
```diff
@@ -1,25 +1,27 @@
...
+GO := /usr/local/go/bin/go
+export GOROOT=/usr/local/go
+export GOPATH=$(HOME)/go
+export GOBIN=$(GOPATH)/bin
 
 all: install build
 
 install: build
-	install -D $(GOPATH)/bin/go-server-server $(DESTDIR)/usr/sbin/go-server-server
+	/usr/bin/install -D $(GOPATH)/bin/go-server-server debian/sonic-rest-api/sbin/go-server-server
```
Comment (anish-n) [inline Makefile]: This and other changes seem to generate the go-server-server at /sbin/ vs /usr/sbin/ that it was originally. The supervisor files would need to be modified to look for go-server-server at sbin in order to run the rest-api server.

File: Dockerfile [hunk 8]
```diff
@@ -22,7 +22,7 @@ RUN dpkg -i /debs/libhiredis0.13_0.13.3-2_amd64.deb \
  && dpkg -i /debs/libnl-route-3-200_3.2.27-2_amd64.deb \
  && dpkg -i /debs/libswsscommon_1.0.0_amd64.deb \
  && dpkg -i /debs/libswsscommon-dev_1.0.0_amd64.deb \
- && dpkg -i /debs/sonic-rest-api_1.0.0_amd64.deb
+ && dpkg -i /debs/sonic-rest-api_1.0.1_amd64.deb
 RUN rm -fr /debs
```
Comment (anish-n) [inline Dockerfile]: Dockerfile.build needs changes to use 1.11+ version of go due to some go version dependency change in go-redis

**Pull request merged**